### PR TITLE
Add Checkbox Component and Registration

### PR DESCRIPTION
**Pull request by robinschaaf**
Adds Register component
Adds Checkbox component
Adds functionality to Input component to accept a label/sub-label, new keyboard style
Add new state for Registration to capture data entry
Add/update theme colors for the Registration button

Adds new tests for Register and Checkbox, and new utils for checking valid email and valid login
Updates snapshots for tests that changed due to updated Input component (SignIn and Input) and reducers test for new Registration initial state

### CODE REVIEW

File: src/components/Input.js [hunk 1]
```diff
@@ -1,47 +1,94 @@
...
-  passwordField: React.PropTypes.bool
+  passwordField: React.PropTypes.bool,
+  keyboardType: React.PropTypes.string,
+  inputStyle: React.PropTypes.string,
+  subLabelText: React.PropTypes.string,
+  addLabel: React.PropTypes.string
 }
 
-Input.defaultProps = { labelText: '', passwordField: false };
+Input.defaultProps = { labelText: '', passwordField: false, required: false };
```
Comment (wgranger) [inline src/components/Input.js]: This is me being a bit pedantic, but maybe keep the style of `defaultProps` as with `propTypes` above (putting one key/value of an object per line. I don't think it's necessarily worth a separate commit for just that purpose, but just for the future.

Comment (robinschaaf) [inline src/components/Input.js]: Ahh, yeah I see

File: src/components/Register.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  }
+
+
+  static renderNavigationBar() {
+    return <NavBar showDrawer={false} showBack={true} title={'Register'} />;
+  }
+
+  render() {
+    const registerDisabled = any(converge(or, [isEmpty, isNil]), [this.props.registration.login, this.props.registration.password, this.props.registration.email])
+    const emailChecked = this.props.registration.global_email_communication
```
Comment (wgranger) [inline src/components/Register.js]: Might be good to give a default prop to `registration`. I can see this throwing an error when trying to look for the `global_email_communication` tag if there isn't a registration prop present.

Comment (robinschaaf) [inline src/components/Register.js]: It has the default value set up in the reducer initial state (you can see in reducers/index.js)

**Comment by wgranger** [comment]
Looks good. Just one small line comment. I like the change in color of the registration button. Also, good use of regex for validations.

File: src/components/Register.js (hunk 2 above)
Comment (wgranger) [inline src/components/Register.js]: Ok. Yes, thought it might be the case that I was missing something with Redux. Thanks.

**Pull request merged**